src/V3ThreadPool.h

```diff
@@ -0,0 +1,99 @@
+// -*- mode: C++; c-file-style: "cc-mode" -*-
+//*************************************************************************
+// DESCRIPTION: Verilator: Thread pool for Verilator itself
+//
+// Code available from: https://verilator.org
+//
+//*************************************************************************
+//
+// Copyright 2005-2022 by Wilson Snyder.  This program is free software; you can
+// redistribute it and/or modify it under the terms of either the GNU
+// Lesser General Public License Version 3 or the Perl Artistic License
+// Version 2.0.
+//
+// Verilator is distributed in the hope that it will be useful,
+// but WITHOUT ANY WARRANTY; without even the implied warranty of
+// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+// GNU General Public License for more details.
+//
+//*************************************************************************
+
+#ifndef _V3THREADPOOL_H_
+#define _V3THREADPOOL_H_ 1
+
+#include "V3Ast.h"
+
+#include <condition_variable>
+#include <functional>
+#include <future>
+#include <list>
+#include <mutex>
+#include <queue>
+#include <thread>
+
+//============================================================================
+
+class V3ThreadPool;
+// The global thread pool
+extern V3ThreadPool v3ThreadPool;
+
+class V3ThreadPool final {
+    // MEMBERS
+    typedef std::function<void()> job_t;
+
+    std::mutex m_mutex;  // Mutex for use by m_queue and m_cv
+    std::queue<job_t> VL_GUARDED_BY(m_mutex) m_queue;  // Queue of jobs
+    std::condition_variable VL_GUARDED_BY(m_mutex) m_cv;  // Conditions to wake up workers
+    std::list<std::thread> m_workers;  // Worker threads
+
+    bool m_shutdown = false;  // Flag signalling termination
+
+public:
+    // CONSTRUCTORS
+    V3ThreadPool() = default;
+
+    ~V3ThreadPool() { resize(0); }
+
+    // METHODS
+    // Resize thread pool to n workers (queue must be empty)
+    void resize(unsigned n);
```

Comment:
```suggestion
    void resize(unsigned n) VL_MT_SAFE;
```

Add VL_MT_SAFE to everything member in the header (assuming everything is safe).